aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -887,6 +905,10 @@ module aptos_framework::fungible_asset {
             metadata,
             amount
         } = fa;
+        event::emit(Burn {
```

Comment:
why are we emitting this in burn_internal? Is this legit?
>     /// CAN ONLY BE CALLED BY coin.move for migration.

Reply:
This simply means the `coin` module is only allowed to call this function, as it is `(friend)` gated.